src/charm.py

```diff
@@ -87,6 +92,115 @@ def _spring_boot_port(self) -> int:
             return port
         return 8080
 
+    def _exec(
+        self, command: typing.List[str], environment: typing.Optional[typing.Dict[str, str]] = None
+    ) -> typing.Tuple[int, str, str]:
+        """Execute a command in Spring Boot application container with a timeout of 60s.
+
+        Args:
+            command: command to be executed.
+            environment: environment variables for the process.
+
+        Returns:
+            A 3-tuple of exit code, stdout and stderr.
+        """
+        container = self._spring_boot_container()
+        process = container.exec(command, environment=environment, timeout=60)
+        try:
+            stdout, stderr = process.wait_output()
+            return 0, stdout, stderr
+        except ExecError as error:
+            return error.exit_code, typing.cast(str, error.stdout), typing.cast(str, error.stderr)
+
+    def _parse_human_readable_units(self, number: str) -> int:
+        """Parse numbers with human-readable units like K, M and G.
+
+        Args:
+            number: input string, something like ``"1G"`` or ``"33m"``.
+
+        Returns:
+            Parsed result, as an integer.
+
+        Raises:
+            ValueError: when the input number is invalid.
+        """
+        number = number.lower()
+        unit = number[-1].lower()
```

Comment:
Wouldn't the `unit` already be `lower` since `number` is `lower` already?

Reply:
Fixed, thanks!